### PR TITLE
Fixes #19623 - add comparison to portkey for vmware networks

### PR DESCRIPTION
**Pull request by wiad**
Details in http://projects.theforeman.org/issues/19623

### CODE REVIEW

**Comment by dLobatog** [comment]
[test upgrade]
[test katello]
[test foreman]

(this triggers our CI test)

File: app/models/concerns/fog_extensions/vsphere/server.rb [hunk 1]
```diff
@@ -40,6 +40,7 @@ def select_nic(fog_nics, nic)
         all_networks = service.raw_networks(datacenter)
         vm_network = all_networks.detect { |network| network._ref == nic_attrs['network'] }
         vm_network ||= all_networks.detect { |network| network.name == nic_attrs['network'] }
+        vm_network ||= all_networks.detect { |network| network.key == nic_attrs['network'] }
```
Comment (dLobatog) [inline app/models/concerns/fog_extensions/vsphere/server.rb]: Nitpick, you may as well put everything on the same line:

```ruby
          vm_network ||= all_networks.detect { |network| [network.name, network.key].include? nic_attrs['network'] } 
```

**Review by dLobatog — CHANGES_REQUESTED** [review]

**Comment by timogoebel** [comment]
@dLobatog: I can't test the issue described in the ticket, but I'll be happy to test this for regressions & merge this next week.

**Review by dLobatog — APPROVED** [review]

**Review by dLobatog — COMMENTED** [review]
@timogoebel Don't worry, I tested just the basic case. The proper fix has to come through fog-vmware though. If tests pass I'll merge.

@wiad Remember to make the commits compliant with @theforeman-bot 's message tips 😉 https://github.com/theforeman/foreman/pull/4611#issuecomment-310384125 It helps with our automation. Don't worry now though, I will just rebase your two commits into one with the right title.

**Comment by wiad** [comment]
@dLobatog  yeah, I forgot the commit message style convention in the last commit but hoped it would be ok anyway - thanks! I'll remember it next time.

**Comment by dLobatog** [comment]
[test upgrade]
[test katello]
[test foreman]

**Comment by dLobatog** [comment]
Thanks @wiad !

**Pull request merged**